lib/chat/utils/DateFormatter.dart

```diff
@@ -0,0 +1,11 @@
+import 'package:intl/intl.dart';
+
+class _Constants {
+  static final dateFormatter = DateFormat('dd MMMM yyyy');
+}
+
+class DateFormatter {
+
+  static String formatDate(DateTime date) => _Constants.dateFormatter.format(date);
```

Comment:
This formatting is being used in several locations. Anyone has a better idea to do so?

Reply:
I don't know at which place we can add that right now, I don't like too much to have that inside `utils` because is where all the things you don't know where to add ends being. @LeeMatthewHiggins in FarmSmart how you handle these cases?

Reply:
not sure I like his global scope class, not sure it adds much. I would just use the standard formatter locally with a constant for now. the refactor can sort out properly. In the refactor the date picker input handler will return the date value picked and the display message, so will handle it all internally.

Reply:
DateFormatter class removed!

Reply:
> I don't know at which place we can add that right now, I don't like too much to have that inside `utils` because is where all the things you don't know where to add ends being. @LeeMatthewHiggins in FarmSmart how you handle these cases?

Date formatting is a presentation layer thing, so it should be left until the last point to convert to a formatted string. And that format will be decided by the design and may be different in different screens. Two screens may share the same format, but we don't want to link these two unrelated screens by having a global format, so its better to have the format decided at the point of use, if it happens to be the same on two screens then we will have the duplicated format(which is small) but I think that's worth it, if the design changes we can easily diverge without having to check all the places that format is shared.